Mod overslot fix

Maximal08 opened the pull request:
## About The Pull Request

close: https://github.com/tgstation/tgstation/issues/90199

Demonstration how it works:

https://github.com/user-attachments/assets/5eea856f-ebfb-4670-8e43-dad32ad01e95

## Why It's Good For The Game
Well it's a bug fix.
## Changelog
:cl:
fix:Fix the overlap of 2 different modsuits on each other
/:cl:

Maximal08 committed:
mod overslot fix

Maximal08 committed:
mod_overslot_fix

Maximal08 commented:
I'm not sure if this is the best possible implementation, so I'm waiting for a review.

Maximal08 commented:
> You currently have a negative Fix/Feature pull request delta of -5. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

Thanks bot, I crying.

Maximal08 committed:
TRAIT_NODROP_fix

SmArtKar commented on code/modules/mod/mod_activation.dm:
```diff
@@ -86,9 +86,21 @@
 			part_datum.overslotting = overslot
 			wearer.transferItemToLoc(overslot, part, force = TRUE)
-			RegisterSignal(part, COMSIG_ATOM_EXITED, PROC_REF(on_overslot_exit))
+			RegisterSignal(part, COMSIG_ATOM_EXITED, PROC_REF(on_overslot_exit), override = TRUE)
```
If you need to use override, you most likely have a flaw in your op order somewhere, exceptions are rare.

SmArtKar commented on code/modules/mod/mod_activation.dm:
```diff
@@ -86,9 +86,21 @@
+				balloon_alert(user, "can't remove [overslot]!")
+				playsound(src, 'sound/machines/scanner/scanbuzz.ogg', 25, TRUE, SILENCED_SOUND_EXTRARANGE)
+				return FALSE
+			if(!overslot.forceMove(src))
+				balloon_alert(user, "failed to store old part!")
+				playsound(src, 'sound/machines/scanner/scanbuzz.ogg', 25, TRUE, SILENCED_SOUND_EXTRARANGE)
+				return FALSE
```
forceMove cannot fail in sane circumstances here, so the check is redundant

SmArtKar reviewed (changes requested):
This is not a proper fix, as discussed in discord a satisfactory solution would be making MOD overslotting into a clothing flag and assigning it to items which we want to get overslotted.

Maximal08 committed:
update_not_a_complete_fix

Maximal08 committed:
Update mod_activation.dm

Maximal08 committed:
Update mod_activation.dm

Maximal08 committed:
Merge branch 'master' into mod_overslot_fix

Maximal08 committed:
Merge branch 'mod_overslot_fix' of https://github.com/Maximal08/tgstation into mod_overslot_fix

Maximal08 committed:
overslotting_using_flags

Maximal08 commented:
> This is not a proper fix, as discussed in discord a satisfactory solution would be making MOD overslotting into a clothing flag and assigning it to items which we want to get overslotted.

I listened to you and was able to figure out the flag system to do this.

SmArtKar commented on code/__DEFINES/clothing.dm:
```diff
@@ -42,3 +42,5 @@
 
 /// How much integrity does a shirt lose every time we bite it?
 #define MOTH_EATING_CLOTHING_DAMAGE 15
+
+#define CLOTHING_MOD_OVERSLOTTING (1<<15) // Allows this clothing item to overslot other items when equipped
```
```suggestion
/// Allows this clothing item to overslot other items when equipped
#define CLOTHING_MOD_OVERSLOTTING (1<<15) 
```
So that autodoc parses it

Maximal08 replied:
I did it.

Maximal08 committed:
autodoc_fix

SmArtKar reviewed (changes requested):
You're only modifying retraction, not deployment - this does nothing. You should only apply this to the overslotting's first check, not the failure part.

Maximal08 committed:
deploy_the_original_version

I'm tired of rolling back this feature for tests, so for now I've returned it to its original state.

Maximal08 committed:
Update mod_activation.dm

Maximal08 commented:
https://github.com/user-attachments/assets/1ff892fd-f3b8-4a2b-9b38-c36e9e06df88

I've updated this, now it works much nicer and cleaner thanks for the advice.
